apps/web/src/components/dashboard/peaks/raw-energy-consumption-card-chart.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+
+import { useCallback, useMemo, useState } from "react";
+import { EnergyPeakDeviceAssignmentDialog } from "./energy-peak-device-assignment-dialog";
+import EnergyConsumptionCardChart from "../energy-consumption-card-chart";
+import React from "react";
+import type { Peak, PeakAssignment } from "@/types/peaks/peak";
+
+interface Props {
+    data: { id: number, energy: number, timestamp: string }[];
+    peaks: PeakAssignment[];
+    devices: { id: number; userId: number; name: string; created: Date | null; }[] | null;
+}
+
+
+/**
+ * Chart for {@link AggregationType.Raw}. Unlike {@link EnergyConsumptionCardChart}, this chart provides further
+ * functionality, i.e. displaying peaks and the ability to assign a device to a peak.
+ */
+export default function RawEnergyConsumptionCardChart({ data, peaks, devices }: Props) {
+    const [open, setOpen] = useState(false);
+    const [value, setValue] = useState<Peak | null>(null);
+
+    const clickCallback = useCallback((callbackData: { id: number, energy: number, timestamp: string | number | undefined, device?: number }) => {
+        setValue({
+            id: Number(callbackData.id),
+            energy: Number(callbackData.energy),
+            timestamp: callbackData.timestamp?.toString() || "",
+            device: callbackData.device ? Number(callbackData.device) : undefined
+        });
+        setOpen(true);
+    }, [setValue, setOpen])
+    const onClick = (devices && devices.length > 0) ? clickCallback : undefined;
+
+    const convertToAxesValue = useCallback((peak: Peak): Record<string, string | number | undefined> => {
+        const sensorData = data.find(x => x.id === peak.id);
+    
+        return {
+            timestamp: sensorData?.timestamp,
+            energy: sensorData?.energy,
+            id: peak.id,
+            device: peak.device
+        };
+    }, [data]);
+
+    const consumptionChart = useMemo(() => (
```

Comment:
Einzige Anmerkung, wollen wir aus dem Energy Line Chart allgemein ein memoized Component machen?
Dann kann es auch an andere Stelle wieder eingebunden werden. Hab es selber nicht ausprobiert wie gut das klappen würde 🤔 

https://react.dev/reference/react/memo

Reply:
Ein allgemeines Line Chart wäre vermutlich echt ganz cool für später. Ich habe evtl. vor den Vergleich zwischen Stromkosten und Abschlag in einem separaten Line Chart anzeigen zu lassen. Dafür wäre ein allgemeines Line Chart zur Verwendung definitiv hilfreich.

Reply:
> Einzige Anmerkung, wollen wir aus dem Energy Line Chart allgemein ein memoized Component machen?
Dann kann es auch an andere Stelle wieder eingebunden werden.

Meinst du damit, dass das EnergyConsumptionCardChart (https://github.com/pgenergy/energyleaf/pull/12/files#diff-a1f61e0c2fb96c9ee08cf7a7507514e44c4e3a63c2d690e8499d78e5c6d10096R19) memorized gemacht werden sollte, das ich hier verwende? Das ist ja so gesehen schon die allgemeine Component dahinter.

> Dafür wäre ein allgemeines Line Chart zur Verwendung definitiv hilfreich.

Ein allgemeines LineChart gibt es ja schon, das du dafür verwenden kannst. Das hier für die Energie würdest du wahrscheinlich ohnehin nicht dafür nutzen können, weil für die Kosten usw. andere Daten erforderlich wären.